dagmc_stats.py

```diff
@@ -300,3 +300,184 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
         coarseness.append(len(surf_area)/sum(surf_area))
 
     return coarseness
+
+
+def get_beta_angles(my_core, pair_tris, common_vert, beta_angles):
+    """
+    Gets the beta angles of given triangle pairs
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    pair_tris : triangle pair
+    common_vert : common vertices in the triangle pair
+    beta_angles: list that stores the beta angles of given triangle pair
+    """
+    vert1 = common_vert[0]
+    vert2 = common_vert[1]
+    side_lengths = []
+    
+    for m in range(2):
+        adj_verts = my_core.get_adjacencies(pair_tris[m], 0, op_type=1)
+        for n in range(3):
+            if adj_verts[n] != vert1 and adj_verts[n] != vert2:
+                other_vert = adj_verts[n]        
+        side_lengths.append(np.linalg.norm(my_core.get_coords(vert1)-my_core.get_coords(vert2)))
+        side_lengths.append(np.linalg.norm(my_core.get_coords(vert1)-my_core.get_coords(other_vert)))
+        side_lengths.append(np.linalg.norm(my_core.get_coords(vert2)-my_core.get_coords(other_vert)))
+        beta_angles.append(np.arccos((side_lengths[1] * side_lengths[1]
+                            + side_lengths[2] * side_lengths[2]
+                            - side_lengths[0] * side_lengths[0])
+                            /(2.0 * side_lengths[1] * side_lengths[2])))
+    
```

Comment:
I wonder if there is a formulation of this that doesn't require you to identify the `other_vert` and reuse your `get_tri_side_lengths` method
```
shared_side_length = np.linalg.norm(my_core.get_coords(vert1) - my_core.get_coords(vert2))
side_lengths = get_tri_side_lengths(my_core, tri)
cos_beta_angle = (np.dot(side_lengths,side_lengths) - 2*shared_side_length*shared_side_length)/
                            (2.0 * np.prod(side_lengths)/shared_side_length)
beta_angles.append(np.arccos(cos_beta_angle))
```

Reply:
furthermore, you might not need to define `vert1` and `vert2`, but just use the `common_vert[0]` and `common_vert[1]` since these are now the only place they are used.

Reply:
I'd like to have a larger software design conversation in person with more people involved on this calculation of the beta angles.